src/sync/pull-request.ts

```diff
@@ -70,42 +60,37 @@ export const getPullRequestTask = async (
 		// Fetch in parallel instead. Given that's an expermient for a single customer, let's not
 		// overcomplicate it too much and limit ourselves to 2 pages.
 		const limitedPageSizeCoef = Math.min(5, pageSizeCoef);
-		const shouldFetchNextPage = pageSizeCoef > 5;
-
-		const prFetchJobs: Promise<any>[] = [];
+		const shouldFetchNextPageInParallel = pageSizeCoef > 5;
 
 		const scaledPageSize = perPage * limitedPageSizeCoef;
-		const scaledCursor = Math.max(1, Math.floor(Number(cursor) / limitedPageSizeCoef));
 
-		prFetchJobs.push(doGetPullRequestTask(
-			logger, gitHubInstallationClient, jiraHost, repository,
+		// Cursor 1, 2, 3, 4, 5 should be mapped to scaled cursor 1;
+		// Cursor 6, 7, 8, 9, 10 shoul be mapped to scaled cursor 2;
+		// etc
+		// Given that the page counter starts from 1, we need to deduct 1 first and then add 1 back to the outcome
+		const scaledCursor = 1 + Math.floor((Number(cursor) - 1) / limitedPageSizeCoef);
+
+		const data = await fetchNextPagesInParallel(
+			shouldFetchNextPageInParallel ? 2 : 1,
 			scaledCursor,
-			scaledPageSize
-		));
-
-		if (shouldFetchNextPage) {
-			logger.info("Fetch second page in parallel");
-			prFetchJobs.push(doGetPullRequestTask(
-				logger, gitHubInstallationClient, jiraHost, repository,
-				scaledCursor + 1,
-				scaledPageSize
-			));
+			(scaledPageNoToFetch) =>
+				doGetPullRequestTask(
+					logger, gitHubInstallationClient, jiraHost, repository,
+					scaledPageNoToFetch,
+					scaledPageSize
+				)
+		);
+		if ((data.edges?.length) || 0 > 0) {
```

Comment:
0 > 0?

Reply:
hehe Yeah, looks like it is always defined. Well spotted!